samples/cpp/macbeth_chart_detection.cpp

```diff
@@ -0,0 +1,150 @@
+                            "vpu: VPU, "
+                            "vulkan: Vulkan, "
+                            "cuda: CUDA, "
+                            "cuda_fp16: CUDA fp16 (half-float preprocess) }");
+
+string keys = param_keys + backend_keys + target_keys;
+
+int main(int argc, char *argv[])
+{
+    CommandLineParser parser(argc, argv, keys);
+    parser.about(about);
+
+    if (parser.has("help"))
+    {
+        cout << about << endl;
+        parser.printMessage();
+        return -1;
+    }
+    string modelName = parser.get<String>("@alias");
+    string zooFile = parser.get<String>("zoo");
+    const char* path = getenv("OPENCV_SAMPLES_DATA_PATH");
+    if ((path != NULL) || parser.has("@alias") || (parser.get<String>("model") != "")) {
+        zooFile = findFile(zooFile);
+    }
+    else{
+        cout<<"[WARN] set the environment variables or pass the arguments --model, --config and models.yml file using --zoo for using dnn based detector. Continuing with default detector.\n\n";
+    }
+
+    keys += genPreprocArguments(modelName, zooFile);
+    parser = CommandLineParser(argc, argv, keys);
+
+    int t = parser.get<int>("type");
+
+    CV_Assert(0 <= t && t <= 2);
+    COLORCHART chartType = COLORCHART(t);
+
+    const string sha1 = parser.get<String>("sha1");
+    const string model_path = findModel(parser.get<string>("model"), sha1);
+    const string config_sha1 = parser.get<String>("config_sha1");
+    const string pbtxt_path = findModel(parser.get<string>("config"), config_sha1);
+
+    int nc = parser.get<int>("num_charts");
+    bool use_gpu = parser.has("use_gpu");
+
+    int waitTime=10;
+    VideoCapture cap;
+    if (parser.has("input"))
+        cap.open(findFile(parser.get<String>("input")));
+    else
+        cap.open(0);
+
+    Ptr<CCheckerDetector> detector;
+    if (model_path != "" && pbtxt_path != ""){
+        Net net = readNetFromTensorflow(model_path, pbtxt_path);
+
+        if (use_gpu)
+        {
+            net.setPreferableBackend(getBackendID("cuda"));
+            net.setPreferableTarget(getTargetID("cuda"));
+        }else{
+            net.setPreferableBackend(getBackendID(parser.get<String>("backend")));
+            net.setPreferableTarget(getTargetID(parser.get<String>("target")));
+        }
+
+        detector = CCheckerDetector::create(net);
+        cout<<"Detecting checkers using neural network."<<endl;
+    }
+    else{
+        detector = CCheckerDetector::create();
+    }
+
+    while (cap.grab())
+    {
+        Mat image, imageCopy;
+        cap.retrieve(image);
+
+        imageCopy = image.clone();
+
+        // Marker type to detect
+        if (!detector->process(image, chartType, nc, true))
+        {
+            printf("ChartColor not detected \n");
+        }
+        else
+        {
+            vector<Ptr<CChecker>> checkers = detector->getListColorChecker();
+            detector->draw(checkers, image);
+
+            Mat chartRgb = checkers[0]->getChartsRGB();
```

Comment:
`getChartsRGB()` 1) returns matrix as a value, 2) that needs to be further reshaped.
`getRefColor()` 1) returns matrix via output parameter 2) that does not need to be reshaped.

please, align API and output shapes in both cases.